test/e2e/app-dir/parallel-routes-use-selected-layout-segment/app/layout.tsx

```diff
@@ -0,0 +1,60 @@
+'use client'
+
+import { useSelectedLayoutSegment } from 'next/navigation'
+import Link from 'next/link'
+
+export default function RootLayout({
+  children,
+  auth,
+  nav,
+}: Readonly<{
+  children: React.ReactNode
+  auth: React.ReactNode
+  nav: React.ReactNode
+}>) {
+  const authSegment = useSelectedLayoutSegment('auth')
+  const navSegment = useSelectedLayoutSegment('nav')
+  const routeSegment = useSelectedLayoutSegment()
+
+  const authSegmentOutput = `${authSegment} (${typeof authSegment})`
+  const navSegmentOutput = `${navSegment} (${typeof navSegment})`
+  const routeSegmentOutput = `${routeSegment} (${typeof routeSegment})`
```

Comment:
what're we using the `typeof` checks for? the test assertions might be a bit clearer to understand if we remove them